pkg/obfuscate/sql_test.go

```diff
@@ -2057,5 +2057,248 @@ func TestToUpper(t *testing.T) {
 			}
 		})
 	}
+}
+
+func TestSQLLexerObfuscation(t *testing.T) {
+	tests := []struct {
+		query            string
+		expected         string
+		replaceDigits    bool
+		dollarQuotedFunc bool
+	}{
+		{
+			query:    "SELECT * FROM users WHERE id = 1",
+			expected: "SELECT * FROM users WHERE id = ?",
+		},
+		{
+			query:         "SELECT * FROM users123 WHERE id = 1",
+			expected:      "SELECT * FROM users? WHERE id = ?",
+			replaceDigits: true,
+		},
+		{
+			query:         "SELECT * FROM users123 WHERE id = 1",
+			expected:      "SELECT * FROM users123 WHERE id = ?",
+			replaceDigits: false,
+		},
+		{
+			query:            "SELECT $func$INSERT INTO table VALUES ('a', 1, 2)$func$ FROM users",
+			expected:         "SELECT $func$INSERT INTO table VALUES (?, ?, ?)$func$ FROM users",
+			dollarQuotedFunc: true,
+		},
+		{
+			query:            "SELECT $func$INSERT INTO table VALUES ('a', 1, 2)$func$ FROM users",
+			expected:         "SELECT ? FROM users",
+			dollarQuotedFunc: false,
+		},
+		{
+			query:            "SELECT * FROM users123 WHERE id = $tag$1$tag$",
+			expected:         "SELECT * FROM users? WHERE id = ?",
+			replaceDigits:    true,
+			dollarQuotedFunc: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run("sqllexer in ObfuscateOnly mode", func(t *testing.T) {
```

Comment:
This works totally fine for these test cases since they are simple, but my previous comment referred to having each test case get a description field. Then this loop would look like:

```suggestion
	for _, tt := range tests {
		t.Run(tt.description, func(t *testing.T) {
```

This would allow test failure messages to point to the specific test case that is failing.